agent/internal/agent.go

```diff
@@ -121,46 +119,23 @@ func (a *Agent) run(ctx context.Context) error {
 	}
 
 	a.log.Tracef("setting up %s runtime", a.opts.ContainerRuntime)
-	var cruntime container.ContainerRuntime
-	switch a.opts.ContainerRuntime {
-	case options.PodmanContainerRuntime:
-		acl, sErr := podman.New(a.opts)
-		if sErr != nil {
-			return fmt.Errorf("failed to build podman client: %w", sErr)
-		}
-		defer func() {
-			if cErr := acl.Close(); cErr != nil {
-				a.log.WithError(cErr).Error("failed to close podman client")
-			}
-		}()
-		cruntime = acl
-	case options.ApptainerContainerRuntime:
-		fallthrough
-	case options.SingularityContainerRuntime:
-		acl, sErr := singularity.New(a.opts)
-		if sErr != nil {
-			return fmt.Errorf("failed to build singularity client: %w", sErr)
-		}
-		defer func() {
-			if cErr := acl.Close(); cErr != nil {
-				a.log.WithError(cErr).Error("failed to close singularity client")
-			}
-		}()
-		cruntime = acl
-	case options.DockerContainerRuntime:
-		dcl, dErr := dclient.NewClientWithOpts(dclient.WithAPIVersionNegotiation(), dclient.FromEnv)
-		if dErr != nil {
-			return fmt.Errorf("failed to build docker client: %w", dErr)
-		}
-		defer func() {
-			a.log.Trace("cleaning up docker client")
-			if cErr := dcl.Close(); cErr != nil {
-				a.log.WithError(cErr).Error("failed to close docker client")
-			}
-		}()
-		cl := docker.NewClient(dcl)
-		cruntime = cl
+	if a.opts.ContainerRuntime != options.DockerContainerRuntime {
+		a.log.Error(a.opts.ContainerRuntime,
+			" Container Runtime is not supported, please update runtime config to use Docker instead.")
+		return fmt.Errorf("%s Container Runtime not available", a.opts.ContainerRuntime)
```

Comment:
nit: I think it's more idiomatic to format the error with the `%s` at the end, so `container runtime not available: %s`

I think Go requires errors to be all lowercase?